### PR TITLE
feat: transaction crud

### PR DESCRIPTION
**Pull request by mathnogueira**
This PR...

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: server/migrations/13_transactions.up.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+);
+
+CREATE TABLE transaction_steps (
+    transaction_id varchar NOT NULL,
+    transaction_version integer NOT NULL,
+    test_id varchar NOT NULL,
+    step_number int NOT NULL,
+
+    CONSTRAINT transaction_steps_transactions_fk
+      FOREIGN KEY (transaction_id, transaction_version) REFERENCES transactions(id, "version")
```
Comment (schoren) [inline server/migrations/13_transactions.up.sql:20]: don't you need an fk here to the tests table?

File: tracetesting/definitions/transaction_create.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      url: ${TARGET_URL}/api/transactions
+      method: POST
+      headers:
+        - key: Content-Type
+          value: application/json
+      body: |
+        {
+            "name": "test-transaction",
+            "description": "a transaction",
+            "steps": []
```
Comment (schoren) [inline tracetesting/definitions/transaction_create.yml:17]: it would be better to have a test here

**Review by schoren — APPROVED** [review]
left 2 nits, but should be good to go!

File: tracetesting/definitions/transaction_create.yml (hunk 2 above)
Comment (schoren) [reply tracetesting/definitions/transaction_create.yml:17]: by test I mean step

File: server/migrations/13_transactions.up.sql (hunk 1 above)
Comment (mathnogueira) [reply server/migrations/13_transactions.up.sql:20]: Thing is that the PK of tests are ìd` and `version`. But as we discussed a couple of days ago, we are always going to use the latest version available. So I decided to not keep the fk here

File: api/openapi.yaml [hunk 3]
```diff
@@ -34,6 +34,139 @@ paths:
                 $ref: "./definition.yaml#/components/schemas/ExecuteDefinitionResponse"
         422:
           description: "trying to create a test with an already existing ID"
+  /transactions:
+    get:
```
Comment (jorgeepc) [inline api/openapi.yaml:38]: Based on the UI design, we are going to need one single endpoint that returns `tests` and `transactions` (should support search, pagination, etc.). We could keep the `get /transactions` endpoint and implement the unified one in a new PR.

File: tracetesting/definitions/transaction_create.yml (hunk 2 above)
Comment (mathnogueira) [reply tracetesting/definitions/transaction_create.yml]: Done

File: server/testdb/postgres.go [hunk 4]
```diff
@@ -107,7 +107,15 @@ func (td *postgresDB) ServerID() (id string, isNew bool, err error) {
 }
 
 func (td *postgresDB) Drop() error {
-	return dropTables(td, "test_runs", "tests", "server", "schema_migrations")
+	return dropTables(
+		td,
+		"transaction_steps",
+		"transactions",
+		"test_runs",
+		"tests",
```
Comment (xoscar) [inline server/testdb/postgres.go:115]: I forgot to add the environments table here, is it necessary?

**Review by xoscar — APPROVED** [review]
Hey @mathnogueira great job man!, I left some non-blocking comments!

File: server/testdb/postgres.go (hunk 4 above)
Comment (mathnogueira) [reply server/testdb/postgres.go:115]: yes, it is. I'll add it

File: server/http/mappings/tests.go [hunk 5]
```diff
@@ -282,6 +298,27 @@ func (m OpenAPI) Runs(in []model.Run) []openapi.TestRun {
 type Model struct {
 	comparators           comparator.Registry
 	traceConversionConfig traces.ConversionConfig
+	testRepository        model.TestRepository
+}
+
+func (m Model) Transaction(ctx context.Context, in openapi.Transaction) (model.Transaction, error) {
+	tests := make([]model.Test, len(in.Steps))
+	for i, testID := range in.Steps {
```
[the remark this answers is not in the document]
Comment (mathnogueira) [reply server/http/mappings/tests.go:306]: I'll try to solve this in another PR

**Pull request merged**